v-next/hardhat/src/internal/cli/telemetry/sentry/reporter.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  HardhatError,
+  HardhatPluginError,
+} from "@ignored/hardhat-vnext-errors";
+
+import { getHardhatVersion } from "../../../utils/package.js";
+import { isTelemetryAllowed } from "../telemetry-permissions.js";
+
+import { getSubprocessTransport } from "./transport.js";
+
+// TODO: replace with PROD version
+// export const SENTRY_DSN =
+//   "https://38ba58bb85fa409e9bb7f50d2c419bc2@o385026.ingest.sentry.io/522486955555555555";
+export const SENTRY_DSN =
+  "https://d578a176729662a28e7a8da268d36912@o385026.ingest.us.sentry.io/4507685793103872"; // DEV
+
+export async function sendErrorTelemetry(error: Error): Promise<boolean> {
+  const instance = await Reporter.getInstance();
+  return instance.reportError(error);
+}
+
+// ATTENTION: this function is exported for testing, do not directly use it in production
+export function _testResetReporter(): void {
+  Reporter.deleteInstance();
+}
+
+class Reporter {
+  // GENERAL EXPLANATION:
+  // 1) The 'reportError' function collects the error and passes it to our custom Sentry transporter.
+  // 2) The custom transporter receives the JavaScript error serialized by Sentry.
+  // 3) This serialized error is then passed to a detached subprocess, which anonymizes all the information before sending it to Sentry.
+
+  static #instance: Reporter | undefined;
+  readonly #telemetryEnabled: boolean;
+
+  private constructor(telemetryAllowed: boolean) {
+    this.#telemetryEnabled = telemetryAllowed;
+  }
+
+  public static async getInstance(): Promise<Reporter> {
+    if (this.#instance !== undefined) {
+      return this.#instance;
+    }
+
+    const telemetryAllowed = await isTelemetryAllowed();
+    this.#instance = new Reporter(telemetryAllowed);
+
+    const { Integrations, init, setExtra } = await import("@sentry/node");
+
+    const linkedErrorsIntegration = new Integrations.LinkedErrors({
+      key: "parent",
```

Comment:
I think this is `cause` now, because in v3 we follow the new js standard

Reply:
[Done here](https://github.com/NomicFoundation/hardhat/pull/5570/commits/bdb9042d15e6896415bbc12393c941cd2d136e08)